spec/services/search_spec.rb

```diff
@@ -86,39 +75,24 @@
       let!(:multi_match_page) {
         create(:page,
                title: 'multimatch page',
-               widgets: [matching_widget, petition_widget],
                language: language,
                campaign: campaign,
                tags: [tag])
       }
-      let(:new_params) {
+      let(:multi_match_params) {
         { search: {
-            content_search: test_text,
+            content_search: 'multimatch page',
             tags: [tag.id],
-            widget_type: ['PetitionWidget'],
             language: language.id
           }
         }
       }
-      let(:new_page_searcher) { Search::PageSearcher.new(new_params) }
-      it 'finds a page that matches the search query by tags, language, widget type and text content' do
+      let(:new_page_searcher) { Search::PageSearcher.new(multi_match_params) }
+      it 'finds a page that matches the search query by tags, language and text content' do
         expect(new_page_searcher.search).to match_array([multi_match_page])
       end
     end
 
   end
 
-  describe 'WidgetSearcher' do
-
-    it 'finds the matching widget based on the search text' do
-      expect(Search::WidgetSearcher.text_widget_search(test_text)).to eq([matching_widget])
-    end
-
-    it 'finds widgets that match the specified widget type' do
-      expect(Search::WidgetSearcher.widget_type_search(['PetitionWidget'])).to match_array(
-                [petition_widget])
-    end
-
-  end
-
 end
```

Comment:
I think this spec could be a bit beefier. Search is something with a lot of possibilities, and I think it e-spec-ially merits a serious spec, especially when you can combine queries and stuff like that.
- I think it's important to include both positive and negative cases - 'when I search for foo nothing is returned, when I search for bar the page is returned'. 
- For the text search, I would add specs for matching just the title, just the body, both, or neither. 
- I would also add some specs that successfully return multiple results, I've had bugs only passing back the last element of the list.
- I would expect the AND / OR behavior to be documented in the spec. It's going to be a best guess decision anyway for now, but the spec serves to illustrate that you've consciously decided and shown that it works one way.
- The multiple match could use a bunch more examples. Right now it matches on all of the fields - what happens if it matches only some of them? What if it matches none of them? What about returning multiple that match on different fields?